provider/resource_kpi_threshold_template.go

```diff
@@ -39,209 +44,431 @@ func kpiThresholdTemplateTFFormat(b *models.Base) (string, error) {
+			"description": schema.StringAttribute{
 				Optional:    true,
 				Description: "User defined description of the entity.",
 			},
-			"adaptive_thresholds_is_enabled": {
-				Type:        schema.TypeBool,
+			"adaptive_thresholds_is_enabled": schema.BoolAttribute{
 				Required:    true,
 				Description: "Determines if adaptive threshold is enabled for this KPI threshold template.",
 			},
-			"adaptive_thresholding_training_window": {
-				Type:        schema.TypeString,
+			"adaptive_thresholding_training_window": schema.StringAttribute{
 				Required:    true,
 				Description: "The earliest time for the Adaptive Threshold training algorithm to run over (latest time is always 'now') (e.g. '-7d')",
 			},
-			"time_variate_thresholds": {
-				Type:        schema.TypeBool,
+			"time_variate_thresholds": schema.BoolAttribute{
 				Required:    true,
 				Description: "If true, thresholds for alerts are pulled from time_variate_thresholds_specification.",
 			},
-			"sec_grp": {
-				Type:        schema.TypeString,
+			"sec_grp": schema.StringAttribute{
 				Required:    true,
 				Description: "The team the object belongs to. ",
 			},
-			"time_variate_thresholds_specification": {
-				Required:    true,
-				Type:        schema.TypeSet,
-				Description: "Data structure for time variate threshold specification.",
-				Elem: &schema.Resource{
-					Schema: getKpiThresholdPolicySchema(),
-				},
-			},
 		},
 	}
 }
 
-func kpiThresholdTemplate(ctx context.Context, d *schema.ResourceData, clientConfig models.ClientConfig) (config *models.Base, err error) {
-	body := map[string]interface{}{}
-	body["objectType"] = "kpi_threshold_template"
-	body["title"] = d.Get("title").(string)
-	body["description"] = d.Get("description").(string)
-	body["adaptive_thresholds_is_enabled"] = d.Get("adaptive_thresholds_is_enabled").(bool)
-	body["adaptive_thresholding_training_window"] = d.Get("adaptive_thresholding_training_window").(string)
-	body["time_variate_thresholds"] = d.Get("time_variate_thresholds").(bool)
-	body["sec_grp"] = d.Get("sec_grp").(string)
-
-	timeVariateThresholdsSpecification := map[string]interface{}{}
-	for _, sourceTimeVariateSpecification_ := range d.Get("time_variate_thresholds_specification").(*schema.Set).List() {
-		sourceTimeVariateSpecification := sourceTimeVariateSpecification_.(map[string]interface{})
-		policies := map[string]interface{}{}
-		for _, sourcePolicy_ := range sourceTimeVariateSpecification["policies"].(*schema.Set).List() {
-			sourcePolicy := sourcePolicy_.(map[string]interface{})
-			policyName := sourcePolicy["policy_name"].(string)
-
-			policies[policyName], err = kpiThresholdPolicyToPayload(sourcePolicy)
-			if err != nil {
-				return nil, err
-			}
-		}
-		timeVariateThresholdsSpecification["policies"] = policies
+func (r *resourceKpiThresholdTemplate) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state modelKpiThresholdTemplate
```

Comment:
Let's call this variable `plan`, since it's set from terraform's plan (and to be consistent with other similar variables)